src/Core/ImageWorkshopLayer.php

```diff
@@ -498,12 +498,15 @@ public function mergeAll()
      * Otherwise, it will be set at 0 and 0
      *
      * @param string $unit Use one of `UNIT_*` constants, "UNIT_PIXEL" by default
-     * @param resource $image
+     * @param resource|object $image
      * @param int $positionX
      * @param int $positionY
      */
-    public function pasteImage($unit = self::UNIT_PIXEL, $image, $positionX = 0, $positionY = 0)
+    public function pasteImage($unit, $image, $positionX = 0, $positionY = 0)
     {
+        if (!in_array($unit, [self::UNIT_PIXEL,self::UNIT_PERCENT])) {
+            $unit = self::UNIT_PIXEL;
```

Comment:
Not sure about this.

I think if the parameter value isn't correct, we should continue the process, we should throw an `InvalidArgumentException`.